### PR TITLE
1.19.3-pre1 variant for predicates

### PR DESCRIPTION
**Pull request by YanisBft**

### CODE REVIEW

**Review by Shnupbups — APPROVED** [review]
Bit annoying to have a mix of `Variant` and `Type`, but I suppose it's best that we keep to the naming used in the entity's respective NBT formats as we currently do until (hopefully) Mojang moves other mobs to a similar registry-backed variant system like Frogs and Cats have.

File: mappings/net/minecraft/entity/passive/FoxEntity.mapping [hunk 1]
```diff
@@ -171,6 +171,7 @@ CLASS net/minecraft/class_4019 net/minecraft/entity/passive/FoxEntity
 		FIELD field_17998 TYPES [Lnet/minecraft/class_4019$class_4039;
 		FIELD field_18000 id I
 		FIELD field_18001 key Ljava/lang/String;
+		FIELD field_41548 TYPE_BY_ID Lnet/minecraft/class_3542$class_7292;
```
Comment (enbrain) [inline mappings/net/minecraft/entity/passive/FoxEntity.mapping]: This is a codec. If anything, `field_17998` is the type-by-id array.
```suggestion
		FIELD field_41548 CODEC Lnet/minecraft/class_3542$class_7292;
```

File: mappings/net/minecraft/entity/passive/MooshroomEntity.mapping [hunk 2]
```diff
@@ -19,6 +19,7 @@ CLASS net/minecraft/class_1438 net/minecraft/entity/passive/MooshroomEntity
 	CLASS class_4053 Type
 		FIELD field_18111 name Ljava/lang/String;
 		FIELD field_18112 mushroom Lnet/minecraft/class_2680;
+		FIELD field_41549 TYPE_BY_ID Lnet/minecraft/class_3542$class_7292;
```
Comment (enbrain) [inline mappings/net/minecraft/entity/passive/MooshroomEntity.mapping]:
```suggestion
		FIELD field_41549 CODEC Lnet/minecraft/class_3542$class_7292;
```

File: mappings/net/minecraft/entity/vehicle/BoatEntity.mapping [hunk 3]
```diff
@@ -88,6 +88,7 @@ CLASS net/minecraft/class_1690 net/minecraft/entity/vehicle/BoatEntity
 	METHOD method_7557 asItem ()Lnet/minecraft/class_1792;
 	CLASS class_1691 Location
 	CLASS class_1692 Type
+		FIELD field_41599 TYPE_BY_ID Lnet/minecraft/class_3542$class_7292;
```
Comment (enbrain) [inline mappings/net/minecraft/entity/vehicle/BoatEntity.mapping]:
```suggestion
		FIELD field_41599 CODEC Lnet/minecraft/class_3542$class_7292;
```

**Review by enbrain — CHANGES_REQUESTED** [review]

**Review by enbrain — APPROVED** [review]

File: mappings/net/minecraft/entity/VariantHolder.mapping [hunk 4]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_7988 net/minecraft/entity/VariantHolder
+	METHOD method_47826 getVariant (Ljava/lang/Object;)V
```
Comment (apple502j) [inline mappings/net/minecraft/entity/VariantHolder.mapping]: :concern:

**Review by apple502j — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/VariantHolder.mapping (hunk 4 above)
Comment (Shnupbups) [reply mappings/net/minecraft/entity/VariantHolder.mapping]:
```suggestion
	METHOD method_47826 setVariant (Ljava/lang/Object;)V
```

**Review by Shnupbups — CHANGES_REQUESTED** [review]

**Review by Shnupbups — APPROVED** [review]

**Review by apple502j — APPROVED** [review]

**Comment by AkashiiKun** [comment]
AxolotlEntity's getVariant and setVariant is missing

method_47826 = setVariant
method_47827 = getVariant

**Pull request merged**